_posts/2019-09-09-what-is-mtls.md

```diff
@@ -4,25 +4,86 @@ current: post
+
+The certificate is then verified by each endpoint according to its signiature which is the provider of the certificate, the CA. With mTLS, it's the exact same except both the server and client verify each other so each endpoint has an x509 certificate. You can think of the CA as vouching for a device's identity.
+
+The actual components of a certificate are determined by a greater structure called PKI (public key infrastructure). PKI is a set of tools that allows for you to encrypt or digitally sign data as with certificates. Within the PKI architecture includes the Certificate Authority which is a trusted third party that signs off on a digital certificate. General services are often signed for by a public CA which will verify any user. As we've said before, WoTT has its own CA but WoTT's CA is private ensuring that only recognised users are signed for. 
+
+Therefore, WoTT uses a PKI structure in order to generate key components of our certificates. At WoTT, we assign each device a unique WoTT ID. This is publically available in your WoTT certificate that is downloaded to your device. If you have the agent installed, you can check this ID with the simple command `wott-agent whoami`. When you have the agent, WoTT will fetch certificates to your device (and these are updated every 7 days. You can check the status of this on your dashboard).
+
+{% asset blog/select-device.png srcset:width="1300 2x" srcset:width="650 1x" alt="WoTT device" %}
+{% asset blog/device-security.png srcset:width="1300 2x" srcset:width="650 1x" alt="security tab" %}
+
+scroll down the security tab to see.
+
+{% asset blog/certificate-expiry.png srcset:width="1300 2x" srcset:width="650 1x" alt="certificate expiry and renewal date" %}
+
+On your node itself, WoTT will have downloaded 4 files.
+
+```
+ca.crt
+client.crt
+client.key
+combined.pem
+```
+The `ca.crt` is the root certificate that is fetched from WoTT. This is what is used to validate your `client.crt` (which contains the device's public key). The other two files contain your private keys and should never leave the device. 
+
+## Why you should care?
+
+For a start, it's probably the most common thing used since SSL became outdated. Several companies use mTLS. For example Skype uses it to secure business servers; and Cloudflare is a large provider of PKI for mTLS. It's particularly important in cases where you want encrypted, secure traffic in *both* directions. Many devices now do not use logins (such as IoT devices) meaning that they would either be blocked from accessing a resource, or access a resource insecurely. Additionally, not using TLS or similar encryption tools without authentication can lead to 'man in the middle attacks.' Using mTLS is a way to provide an identity to a device/server that can be cryptographically verified. In other words, it's a way of making your resources more flexible and available whilst still being secure through authentication.
```

Comment:
Cloudflare offers something called [Cloudflare Access](https://developers.cloudflare.com/access/service-auth/mtls/) but to my knowledge, you can't use *their* PKI for this.